packages/cubejs-schema-compiler/test/integration/postgres/yaml-compiler.test.ts

```diff
@@ -482,6 +482,65 @@ cubes:
     );
   });
 
+  it('should correctly handle extensions with joins', async () => {
+    const { compiler, joinGraph, cubeEvaluator } = prepareYamlCompiler(`
+      cubes:
+        - name: base_users
+          sql: "SELECT 1"
+
+          dimensions:
+            - name: time
+              sql: "{CUBE}.timestamp"
+              type: time
+            - name: user_id
+              sql: "user_id"
+              type: number
+              primary_key: true
+              public: true
+
+          joins:
+            - name: base_user_joins
+              sql: "{CUBE}.user_id = {base_user_joins}.user_id"
```

Comment:
What would/should happen if the current cube is referenced by its name instead of `CUBE` here? If it fails, would the error be comprehensible enough?

Reply:
Also, if the error is not very comprehensible and it's not that easy to fix, it might help to leave a note on join inheritance on this page: https://cube.dev/docs/product/data-modeling/concepts/code-reusability-extending-cubes